ksy_schema.json

```diff
@@ -243,19 +243,23 @@
         },
         "terminator": {
           "type": "integer",
-          "description": "string reading will stop when it encounters this value default is 0"
+          "description": "string reading will stop when it encounters this value\n\ndefault is 0",
+          "default": 0
```

Comment:
This default is only true for `type: strz`. For `type: str`, specifying no `terminator` means that the string is always read fully based on its `size`.

I think it would be best to remove the `"default"` key from the schema (unless there's a good way to express this relationship between `type` and the default value?) and explain this in text form.

Reply:
Good point. I also think that the description should state that default value is 0 only if the type is `strz`. Otherwise this default doesn't make any sense.

When you explicitly set a property to the default value, nothing should change, because this value is implicitly considered. This is not true if type is `str`, because it **does** matter if you explicitly specify `terminator: 0` or not.

But we may go from the other side. I would rather ban use of `terminator` with `type: strz` completely: it is weird when you do:
```yaml
  - id: name
    type: strz
    terminator: 0x2e
```
The `type: strz` is just shortcut for `type: str`, `terminator: 0x00`, right? So why not write it like this:
```yaml
  - id: name
    type: str
    terminator: 0x00
    terminator: 0x2e
```

My point is, that when you override `type: strz` (zero-terminated string) with `terminator: 0x2e` or anything else, it is no **zero**-terminated string anymore. This is usually an oversight and hence I think the schema validator should complain. On top of that, the default value of `terminator` only brings confusion because it simply doesn't apply in all cases.

Reply:
> I would rather ban use of `terminator` with `type: strz` completely

Good idea - completely agree with all of your points. I can't think of a case where `type: strz` with `terminator` is a useful combination. (I assume that some people will be using this combination, but probably only by accident, in which case it's even better to warn about this.)